modules/test-utils/src/generate-layer-tests.js

```diff
@@ -135,24 +138,51 @@ function makeAltPropTestCase(propName, propTypes, defaultProps) {
       } else {
         newProps[propName] = defaultProps[propName] + 1;
       }
-      title = String(newProps[propName]);
-      break;
+      return [
+        {
+          title: `${propName}: ${String(newProps[propName])}`,
+          props: newProps
+        }
+      ];
 
-    case 'accessor':
+    case 'accessor': {
       if (typeof defaultProps[propName] === 'function') {
         return null;
       }
-      newProps[propName] = () => defaultProps[propName];
+      let callCount = 0;
+      newProps[propName] = () => {
+        callCount++;
+        return defaultProps[propName];
+      };
       newProps.updateTriggers = {
-        [propName]: propName
+        [propName]: 'function'
       };
-      title = `() => ${defaultProps[propName]}`;
-      break;
+      const onBeforeUpdate = () => (callCount = 0);
+      const onAfterUpdate = () => assert(callCount > 0, 'accessor function is called');
+
+      return [
+        {
+          title: `${propName}: () => ${defaultProps[propName]}`,
+          props: newProps,
+          onBeforeUpdate,
+          onAfterUpdate
+        },
+        {
+          title: `${propName}: updateTrigger`,
+          updateProps: {
+            updateTriggers: {
+              [propName]: 'function+trigger'
+            }
+          },
+          onBeforeUpdate,
+          onAfterUpdate
+        }
+      ];
+    }
```

Comment:
Nit: Do we care about adding test case, where `updateTrigger` doesn't change and `callCount` should be 0.

Reply:
I don't think that needs to be tested per layer. Theoretically that should be covered by the unit tests of `diffProps` and `AttributeManager`.